department-of-veterans-affairs/caseflow
# Case Details - Assign Veteran and complete task.

Resolves #7072 

### Description
Integrates hearings and Queue's Case Details. Created a Schedule Hearing task with an assign hearing action. Expanded the task model to have many business payload items. Promoted the Regional Office selector to common components and defined a common reducer for any common component to have. Added this common reducer to both hearings and queue reducers. 

### Acceptance Criteria
- [ ] Test cases pass

### Testing Plan
1. Go to Assign Hearings page.
2. Click on a veteran ready to get a hearing scheduled.
3. Verify you go to Case Details and see a task with the Assign Hearing action.
4. When Assign Hearing is selected a modal pops up with the hearings info. 
5. Click on Assign in the modal.
6. Confirm the veteran now shows up in the Schedule Hearings tab in Assign Hearings.
7. Check the database that the hearsched row was created and that the location is now either 38 or 36 depending on whether it is a video or central office hearing.
